finite_difference/src/field_mod.f90

```diff
@@ -187,7 +198,10 @@ function r2d_field_constructor(grid,    &
        ntiley = 1
     end if
     nthreads = 1
-!$  nthreads = omp_get_max_threads()
+    if (present(do_tile) .and. do_tile) then
```

Comment:
I got caught out by this recently - Fortran does not do lazy evaluation so you can't assume that the `present(do_tile)` will happen before the reference to `do_tile`. It's a pain but you have to have two, separate `if` statements. (I was getting away with it until I tried the PGI compiler.)

Reply:
Gee, 30+years of Fortran programming, and I didn't know - embarrassing. I guess this kind of construct is more common in C++ ;)

Reply:
Don't worry - I only discovered this myself within the last month!